pkg/tool/cmdexecutor.go

```diff
@@ -0,0 +1,35 @@
+package tool
+
+import (
+	"strings"
+	"text/template"
+
+	"github.com/helm/chart-testing/v3/pkg/exec"
+	"github.com/mattn/go-shellwords"
+)
+
+type CmdTemplateExecutor struct {
+	exec exec.ProcessExecutor
+}
+
+func NewCmdTemplateExecutor(exec exec.ProcessExecutor) CmdTemplateExecutor {
+	return CmdTemplateExecutor{
+		exec: exec,
+	}
+}
+
+func (t CmdTemplateExecutor) RunCommand(cmdTemplate string, data interface{}) error {
+	var template = template.Must(template.New("command").Parse(cmdTemplate))
+	var b strings.Builder
+	if err := template.Execute(&b, data); err != nil {
+		return err
+	}
+	rendered := b.String()
+
+	words, err := shellwords.Parse(rendered)
+	if err != nil {
+		return err
+	}
+	name, args := words[0], words[1:]
```

Comment:
You should check the length of the slice. There might not be any args.

Reply:
That's not an issue. `words[1:]` will be an empty slice in that case. I added a unit test for that:
https://github.com/torstenwalter/chart-testing/blob/additional-commands/pkg/tool/cmdexecutor_test.go#L29-L39